music.py

```diff
@@ -411,56 +426,68 @@ def get_all_guitar_chords(
+        im = plt.imread(os.path.join(PROJ_DIR, 'static', 'treble_clef.png'))
+        ax.imshow(im, extent=(1, 5, -1, 12))
+        im = plt.imread(os.path.join(PROJ_DIR, 'static', 'bass_clef.png'))
+        ax.imshow(im, extent=(1, 6, -9, -2))
         # staff
         for line in range(2, 12, 2):
             ax.plot(xlim, [line] * 2, 'k-')
-        if self.lowest_line < 2:
-            for line in range(self.lowest_line, 2, 2):
-                ax.plot([note_pos - 2 * note_rad, note_pos + 2 * note_rad], [line] * 2, 'k-')
-        if self.highest_line > 10:
-            for line in range(12, self.highest_line + 2, 2):
-                ax.plot([note_pos - 2 * note_rad, note_pos + 2 * note_rad], [line] * 2, 'k-')
-        shift = 0
-        for note, gap in zip(self.notes, self.gaps):
-            if shift == 0 and gap is not None and gap < 2:
-                shift = 1.75 * note_rad
-            else:
-                shift = 0
-            note_pos_ = note_pos + shift
-            ax.add_patch(plt.Circle(xy=(note_pos_, note.staff_line), radius=0.9 * note_rad, facecolor="none", edgecolor='k'))
-            ax.annotate(note.modifier, xy=(note_pos_ - 2.25 * note_rad, note.staff_line - 0.6), fontsize=12, family='arial')
+        for line in range(-2, -12, -2):
+            ax.plot(xlim, [line] * 2, 'k-')
+        for chord, (lowest_line, highest_line), note_pos in zip(self.chords, self.ledger_lines, note_positions):
+            if lowest_line < -10:
+                for line in range(lowest_line, 10, 2):
+                    ax.plot([note_pos - 2 * note_rad, note_pos + 2 * note_rad], [line] * 2, 'k-')
+            if highest_line > 10:
+                for line in range(12, highest_line + 2, 2):
+                    ax.plot([note_pos - 2 * note_rad, note_pos + 2 * note_rad], [line] * 2, 'k-')
+            shift = 0
+            for note, gap in zip(chord.notes, chord.staff_line_gaps):
+                if shift == 0 and gap is not None and gap < 2:
+                    shift = 1.75 * note_rad
+                else:
+                    shift = 0
+                note_pos_ = note_pos + shift
+                ax.add_patch(plt.Circle(xy=(note_pos_, note.staff_line), radius=0.9 * note_rad, facecolor="none", edgecolor='k'))
+                ax.annotate(note.modifier, xy=(note_pos_ - 2.25 * note_rad, note.staff_line - 0.6), fontsize=12, family='arial')
+                if note.staff_line == 0:
+                    ax.plot([note_pos_ - 2 * note_rad, note_pos_ + 2 * note_rad], [0, 0], 'k-')
         ax.set_aspect(0.9)
         ax.axis('off')
-        fig.savefig(path)
+        plt.tight_layout()
+        fig.savefig(path, bbox_inches='tight', pad_inches=0)
 
 
 class GuitarPosition:
 
-    def __init__(self, positions: dict[Hashable, int], guitar: 'Guitar' = None, max_fret_span: int = DEFAULT_MAX_FRET_SPAN):
+    def __init__(
+            self,
+            positions: dict[Hashable, int],
+            *,
+            notes: Optional[list['Note']] = None,
```

Comment:
This needed to be added for the `GuitarPosition.chord` attribute to retain the correct enharmonic bias. Will clarify with docstrings in a followup